File: Leetcode/1137 - N-th Tribonacci Number/solution.py
def tribonacci(self, n: int) -> int:
    if n == 0:
        return 0
    if n <= 2:
        return 1

    a, b, c = 0, 1, 1
    for _ in range(3, n + 1):
        res = a + b + c
        a, b, c = b, c, res

    return c

File: Leetcode/1137 - N-th Tribonacci Number/solution.cpp
int tribonacci(int n) {
    if (n == 0) return 0;
    if (n <= 2) return 1;

    int a = 0, b = 1, c = 1, res;
    for (int i = 3; i <= n; i++) {
        res = a + b + c;
        a = b;
        b = c;
        c = res;
    }
    
    return c;
}